src/main/scala/io/shiftleft/js2cpg/preprocessing/TranspilingEnvironment.scala

```diff
@@ -24,15 +49,15 @@ object TranspilingEnvironment {
   val NPM: String  = ExternalCommand.toOSCommand("npm")
 
   val YARN_ADD: String =
-    s"$YARN --prefer-offline --ignore-scripts --legacy-peer-deps --dev -W add"
+    s"$YARN --ignore-scripts --legacy-peer-deps --dev -W add"
   val YARN_INSTALL: String =
-    s"$YARN --prefer-offline --ignore-scripts --legacy-peer-deps install"
+    s"$YARN --ignore-scripts --legacy-peer-deps install"
   val PNPM_ADD: String =
-    s"$PNPM --prefer-offline --ignore-scripts add -D"
+    s"$PNPM --ignore-scripts add -D"
   val PNPM_INSTALL: String =
-    s"$PNPM --prefer-offline --ignore-scripts install"
+    s"$PNPM --ignore-scripts install"
   val NPM_INSTALL: String =
-    s"$NPM --prefer-offline --no-audit --progress=false --ignore-scripts --legacy-peer-deps --save-dev install"
+    s"$NPM --no-audit --progress=false --ignore-scripts --legacy-peer-deps --save-dev install"
```

Comment:
what's the effect of removing `--prefer-offline`? (The NPM docs are quite vague.)

Reply:
npm tries to find versions in the local cache and throws if it can't find anything there. In a fresh system or in a CI environment that would lead to failures when requesting specific versions that have never been downloaded.